### PR TITLE
Add flipped curved

### LINKED ISSUE #2137: Curves could use an inverse operator or possible an InverseCurve that inverts a curve.

### PR DESCRIPTION
**Pull request by apwilson**
Fixes https://github.com/flutter/flutter/issues/2137

### CODE REVIEW

File: packages/flutter/lib/src/animation/curves.dart [hunk 1]
```diff
@@ -103,6 +109,20 @@ double _bounce(double t) {
...
+  final Curve curve;
+  InvertedCurve(this.curve);
+  double transform(double t) => curve(1.0 - t);
+}
+
+/// A curve that reverses its given curve.
+class ReversedCurve implements Curve {
+  final Curve curve;
+  ReversedCurve(this.curve);
+  double transform(double t) => 1.0 - curve(t);
```
Comment (abarth) [inline packages/flutter/lib/src/animation/curves.dart]: This isn't legit.  A curve has to map 0.0 to 0.0 and 1.0 to 1.0.

File: packages/flutter/lib/src/animation/curves.dart [hunk 2]
```diff
@@ -103,6 +109,20 @@ double _bounce(double t) {
   return 7.5625 * t * t + 0.984375;
 }
 
+/// A curve that inverts its given curve.
+class InvertedCurve implements Curve {
+  final Curve curve;
+  InvertedCurve(this.curve);
+  double transform(double t) => curve(1.0 - t);
```
Comment (abarth) [inline packages/flutter/lib/src/animation/curves.dart]: This isn't legit either.  If curve maps 0.0 to 0.0 and 1.0 to 1.0, this new curve will violate that invariant.

**Comment by abarth** [comment]
These requirements are documented on line 15 of this file.

**Comment by apwilson** [comment]
Done.

File: packages/flutter/lib/src/animation/curves.dart [hunk 3]
```diff
@@ -103,6 +107,13 @@ double _bounce(double t) {
   return 7.5625 * t * t + 0.984375;
 }
 
+/// A curve that is the reversed inversion of its given curve.
+class FlippedCurve implements Curve {
+  final Curve curve;
+  FlippedCurve(this.curve);
```
Comment (abarth) [inline packages/flutter/lib/src/animation/curves.dart]: Constructors first.

File: packages/flutter/lib/src/animation/curves.dart [hunk 4]
```diff
@@ -103,6 +107,13 @@ double _bounce(double t) {
...
+  double transform(double t) => 1.0 - curve(1.0 - t);
```
Comment (abarth) [inline packages/flutter/lib/src/animation/curves.dart]: `curve.transform`

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/animation/curves.dart (hunk 3 above)
Comment (apwilson) [inline packages/flutter/lib/src/animation/curves.dart]: done

File: packages/flutter/lib/src/animation/curves.dart (hunk 4 above)
Comment (apwilson) [inline packages/flutter/lib/src/animation/curves.dart]: oops, done

**Comment by abarth** [comment]
LGTM

**Pull request merged**